urnx/validate.go

```diff
@@ -0,0 +1,24 @@
+package urnx
+
+import (
+	"regexp"
+)
+
+func validateNamespace(namespace string) (bool, error) {
+	rx, err := regexp.MatchString("^[A-Za-z0-9-]{1,30}$", namespace)
```

Comment:
I'd recommend using [`regexp.MustCompile`][regexp] in `init` for this and `validateResourceType`.

[regexp]: https://pkg.go.dev/regexp#MustCompile

Reply:
✅